### PR TITLE
Add tutorial to restore rds db snapshot to dbinstance/cluster

### PR DESCRIPTION
**Pull request by brucegucode**
Issue #, if available:

Description of changes:

Add tutorial for restore rds snapshot. 
either restore Snapshot to DBCluster or restore DBSnapshot to DBInstance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/content/docs/tutorials/rds-example.md [hunk 1]
```diff
@@ -256,6 +256,50 @@ spec:
 EOF
 ```
 
+## Restore Snapshot to DBInstance or DBCluster
```
Comment (jkatz) [inline docs/content/docs/tutorials/rds-example.md:259]: I would suggest "Restore a Database Snapshot" as the title. We can cover the semantics of `DBInstace` v `DBCluster` in the language below.

File: docs/content/docs/tutorials/rds-example.md [hunk 2]
```diff
@@ -256,6 +256,50 @@ spec:
...
+
+You can restore a snapshot to DBInstance or DBCluster with setting `SnapshotIdentifier` inside `DBCluster` or `DBSnapshotIdentifier` inside `DBInstance` CRD to restore the snapshot to a DBCluster or DBInstance. 
+`SnapshotIdentifier` need match the existing DBCluster snapshot identifier or ARN of DBInstance snapshot, `DBSnapshotIdentifier` need match the identifier of an existing DBSnapshot. 
+Once it's set and resource created, update them will have no effect. You can use the following example to restore a DBSnapshot to DBCluster:
```
Comment (jkatz) [inline docs/content/docs/tutorials/rds-example.md:263]:
```suggestion
You can also restore a database snapshot to a specific `DBInstance` or `DBCluster` using the ACK for RDS controller.

To restore a database snapshot to a `DBInstance`, you must set the `DBSnapshotIdentifier` parameter. `DBSnapshotIdentifier` need match the identifier of an existing DBSnapshot.

To restore a database snapshot to a `DBCluster`, you must set the `SnapshotIdentifier`. The value of `SnapshotIdentifier` must match either an existing `DBCluster` snapshot identifier or an ARN of a `DBInstance`snapshot.

The following examples show how you can restore database snapshots both to `DBCluster` and `DBInstance` resources: 
```

File: docs/content/docs/tutorials/rds-example.md [hunk 3]
```diff
@@ -256,6 +256,50 @@ spec:
...
+cat <<EOF > rds-restore-dbcluster-snapshot.yaml
+apiVersion: rds.services.k8s.aws/v1alpha1
+kind: DBCluster
+metadata:
+  name: "${RDS_CLUSTER_NAME}"
+spec:
+  dbClusterIdentifier: "${RDS_CLUSTER_NAME}"
+  engine: aurora-postgresql
+  engineVersion: "13.7"
+  snapshotIdentifier: arn:aws:rds:${RDS_REGION}:${RDS_CUSTOMER_ACCOUNT}:snapshot:${RDS_DB_SNAPSHOT_IDENTIFIER}
```
Comment (jkatz) [inline docs/content/docs/tutorials/rds-example.md:275]: Where is `RDS_DB_SNAPSHOT_IDENTIFIER` defined?

File: docs/content/docs/tutorials/rds-example.md [hunk 4]
```diff
@@ -256,6 +256,50 @@ spec:
...
+  name: "${RDS_INSTANCE_NAME}"
+spec:
+  allocatedStorage: 20
+  dbInstanceClass: db.m5.large
+  dbInstanceIdentifier: "${RDS_INSTANCE_NAME}"
+  engine: postgres
+  engineVersion: "14"
+  masterUsername: "postgres"
+  multiAZ: true
+  dbSnapshotIdentifier: "${RDS_DB_SNAPSHOT_IDENTIFIER}"
```
Comment (jkatz) [inline docs/content/docs/tutorials/rds-example.md:297]: Where is Where is `RDS_DB_SNAPSHOT_IDENTIFIER` defined?

File: docs/content/docs/tutorials/rds-example.md (hunk 2 above)
Comment (jkatz) [reply docs/content/docs/tutorials/rds-example.md:263]: I think we also need a line on where you can find the snapshot identifier.

**Review by jkatz — CHANGES_REQUESTED** [review]

File: docs/content/docs/tutorials/rds-example.md [hunk 5]
```diff
@@ -256,6 +256,50 @@ spec:
 EOF
 ```
 
+## Restore Snapshot to DBInstance or DBCluster
+
+You can restore a snapshot to DBInstance or DBCluster with setting `SnapshotIdentifier` inside `DBCluster` or `DBSnapshotIdentifier` inside `DBInstance` CRD to restore the snapshot to a DBCluster or DBInstance. 
+`SnapshotIdentifier` need match the existing DBCluster snapshot identifier or ARN of DBInstance snapshot, `DBSnapshotIdentifier` need match the identifier of an existing DBSnapshot.
```
Comment (jaypipes) [inline docs/content/docs/tutorials/rds-example.md:262]:
```suggestion
`SnapshotIdentifier` should match an existing DBCluster snapshot identifier or  the ARN of a DBInstance snapshot, `DBSnapshotIdentifier` should match the identifier of an existing DBSnapshot. 
```

File: docs/content/docs/tutorials/rds-example.md (hunk 2 above)
Comment (jaypipes) [inline docs/content/docs/tutorials/rds-example.md:263]:
```suggestion
Once it's set and the resource is created, updating the `SnapshotIdentifer` or `DBSnapshotIdentifier` fields will have no effect.

Use the following example to to create a DBCluster from a DBSnapshot:
```

File: docs/content/docs/tutorials/rds-example.md [hunk 6]
```diff
@@ -256,6 +256,50 @@ spec:
...
+  dbClusterIdentifier: "${RDS_CLUSTER_NAME}"
+  engine: aurora-postgresql
+  engineVersion: "13.7"
+  snapshotIdentifier: arn:aws:rds:${RDS_REGION}:${RDS_CUSTOMER_ACCOUNT}:snapshot:${RDS_DB_SNAPSHOT_IDENTIFIER}
+EOF
+
+kubectl apply -f rds-restore-dbcluster-snapshot.yaml
+```
+
+Example to restore a DBSnapshot to DBInstance:
```
Comment (jaypipes) [inline docs/content/docs/tutorials/rds-example.md:281]:
```suggestion
Use the following example to create a DBInstance from a DBSnapshot:
```

**Review by jaypipes — CHANGES_REQUESTED** [review]
Few little grammar fixes, otherwise this is great, thank you @brucegucode!

File: docs/content/docs/tutorials/rds-example.md [hunk 7]
```diff
@@ -256,6 +256,62 @@ spec:
 EOF
 ```
 
+## Restore a Database Snapshot
+
+You can also restore a database snapshot to a specific `DBInstance` or `DBCluster` using the ACK for RDS controller.
+
+To restore a database snapshot to a `DBInstance`, you must set the `Spec.DBSnapshotIdentifier` parameter. `Spec.DBSnapshotIdentifier` should match the identifier of an existing DBSnapshot.
```
Comment (jkatz) [inline docs/content/docs/tutorials/rds-example.md:263]: Looking at this again, if we use the `spec` prefix we should follow the convention from earlier in the tutorial:

`spec.dbSnapshotIdentifier`.

Comment (brucegucode) [reply docs/content/docs/tutorials/rds-example.md]: I will follow the convention to remove the `Spec`here

**Comment by jkatz** [comment]
@jaypipes I agree with the sentiment. The wording in question was to ensure that we could clearly identify the scenario for the user.

Going to the RDS docs, it appears your wording does [mirror what is suggested](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Tutorials.RestoringFromSnapshot.html) so I think we should be OK with "Create a database from a DB snapshot" or something equivalent ("Create a database cluster/instance from a DB snapshot").

I still think the header should use the descriptive terms instead of the resource names directly, but agree it should be declarative.

**Comment by brucegucode** [comment]
I agree with being more declarative here, I'm now using word 'Create a Database from Snapshot'

**Review by jaypipes — APPROVED** [review]
OK, I'll compromise :) I'm not thrilled with the usage of "restore a database snapshot" phrasing in the documentation but at least the section title now follows a more declarative style.

**Review by jkatz — APPROVED** [review]

**Pull request merged**